### PR TITLE
Add tests for change_log_level

### PR DESCRIPTION
**Pull request by MorrisLaw**
This PR is a follow-up for comments left on [PR #557](https://github.com/istio/ztunnel/pull/557).

The following changes were made:
- changed level for `"Parsed level: {:?}"` message to debug instead of info. cc/ @keithmattix 
- changed status code for other error in `change_log_level` to also be BAD_REQUEST. cc/ @howardjohn 
- add unit tests for `change_log_level`. cc/ @GregHanson

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]
thanks!

File: src/admin.rs [hunk 1]
```diff
@@ -361,11 +361,11 @@ fn change_log_level(reset: bool, level: &str) -> Response<Full<Bytes>> {
     match tracing::level_filters::LevelFilter::from_str(level) {
         Ok(level_filter) => {
             // Valid level, continue processing
-            tracing::info!("Parsed level: {:?}", level_filter);
+            tracing::debug!("Parsed level: {:?}", level_filter);
```
Comment (GregHanson) [inline src/admin.rs:364]: nit: this API isn't going to be called often (outside of debugging), could probably leave at `info` level

**Review by GregHanson — COMMENTED** [review]
lgtm - thanks for the tests!

File: src/admin.rs (hunk 1 above)
Comment (MorrisLaw) [reply src/admin.rs]: cc/ @keithmattix since he suggested the change initially. Any strong opinions against leaving as info?

Comment (keithmattix) [reply src/admin.rs]: No strong opinions; we can always lower the level later. Feel free to change it back to info

**Comment by nmittler** [comment]
@MorrisLaw these tests are failing on CI: https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_ztunnel/536/test_ztunnel/1671993444202975232/build-log.txt.

Can you rollback? It's blocking my change. Thanks!

**Pull request merged**